ibis/pandas/aggcontext.py

```diff
@@ -313,6 +314,92 @@ def compute_window_spec_interval(_, expr):
     return pd.tseries.frequencies.to_offset(value)
 
 
+def _window_agg_built_in(
+    frame, windowed, function, max_lookback, *args, **kwargs
+):
+    """Apply window aggregation with built-in aggregators.
+    """
+    assert isinstance(function, str)
+    method = operator.methodcaller(function, *args, **kwargs)
+
+    if max_lookback is not None:
+        agg_method = method
+
+        def sliced_agg(s):
+            return agg_method(s.iloc[-max_lookback:])
+
+        method = operator.methodcaller('apply', sliced_agg, raw=False)
+
+    result = method(windowed)
+    index = result.index
+    result.index = pd.MultiIndex.from_arrays(
+        [frame.index]
+        + list(map(index.get_level_values, range(index.nlevels))),
+        names=[frame.index.name] + index.names,
+    )
+    return result
+
+
+def _window_agg_udf(
+    grouped_data, windowed, function, dtype, max_lookback, *args, **kwargs
+):
+    """Apply window aggregation with UDFs.
+    """
+    # Use custom logic to computing rolling window UDF instead of
```

Comment:
include this in the doc-string under Notes

Reply:
Done